python/tester/upload.py

```diff
@@ -153,6 +153,38 @@ def main():
         _log.error(f"No raw files found for {instrument}, aborting.")
 
 
+def _add_to_raw_pool(raw_pool, snap_num, visit, blob):
+    """Add a detector-snap to the raw pool for uploading.
+
+    Parameters
+    ----------
+    raw_pool : mapping [`str`, mapping [`int`, mapping [`activator.FannedOutVisit`, `s3.ObjectSummary`]]]
+        Available raws as a mapping from group IDs to a mapping of snap ID.
+        The value of the innermost mapping is the observation metadata for
+        each detector, and a Blob representing the image taken in that
+        detector-snap.
```

Comment:
We really ought to get a proper class for this! 😂 Maybe I'll find an opportunity...